auth0/data_auth0_client.go

```diff
@@ -0,0 +1,52 @@
+package auth0
```

Comment:
I'm actually using both of these data sources in a real setting and they work great.  Being able to look up a client by its name has been super useful.